map/src/main/kotlin/edu/artic/map/rendering/ObjectsMapItemRenderer.kt

```diff
@@ -72,19 +85,35 @@ class ObjectsMapItemRenderer(private val objectsDao: ArticObjectDao)
             existing: MarkerHolder<MapItemModel>,
             visibleRegion: VisibleRegion,
             mapChangeEvent: MapChangeEvent) {
+        adjustVisibleMarker(existing, null, visibleRegion, mapChangeEvent)
+    }
+
+    private fun adjustVisibleMarker(
+            existing: MarkerHolder<MapItemModel>,
+            selectedArticObject: ArticObject?,
+            visibleRegion: VisibleRegion,
+            mapChangeEvent: MapChangeEvent) {
         val position = getAdjustedLocationFromItem(existing.item)
         val meta = existing.marker.metaData()
-                ?: MarkerMetaData(existing.item, loadedBitmap = false, requestDisposable = null)
+                ?: MarkerMetaData(existing.item, loadedBitmap = false, requestDisposable = null, isSelected = false)
+
+        val isSelected = existing.item.isSelected(selectedArticObject)
 
         if (mapChangeEvent.displayMode !is MapDisplayMode.CurrentFloor
-                || position.isCloseEnoughToCenter(visibleRegion.latLngBounds)) {
+                || (position.isCloseEnoughToCenter(visibleRegion.latLngBounds) || isSelected)) {
             if (!meta.loadedBitmap) {
                 // show loading while its loading.
                 existing.marker.setIcon(loadingBitmap)
 
                 // enqueue replacement and add to marker meta.
                 existing.marker.tag = meta.copy(
-                        requestDisposable = enqueueBitmapFetch(item = existing.item, mapChangeEvent = mapChangeEvent))
+                        requestDisposable = enqueueBitmapFetch(item = existing.item, mapChangeEvent = mapChangeEvent), isSelected = isSelected)
+            } else if (!meta.isSelected && isSelected) {
```

Comment:
Isn't this `if` block doing the same as the next one? You could use the `kotlin.Boolean` infix `xor` operator.